tests/ssl-opt.sh

```diff
@@ -8874,6 +8877,9 @@ run_test    "TLS 1.3: minimal feature sets - gnutls" \
             -c "HTTP/1.0 200 OK"
 
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL
+requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
```

Comment:
```suggestion
```
Not needed for this test

Reply:
I just remember that Roald want to enable COMPATIBILITY_MODE in most of the cases by default, only some special cases to make the disable_COMPATIBILITY_MODE fail.  I am not sure whether this is the intention.

Reply:
Yes but ok to remove it from here.

Reply:
The test case does not depend on MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE